tests/test_dummy.py

```diff
@@ -0,0 +1,309 @@
+from rdkit import Chem
+
+from timemachine.fe.dummy import (
+    enumerate_anchor_groups,
+    identify_anchor_groups,
+    identify_anchors,
+    identify_dummy_groups,
+    enumerate_dummy_ixns,
+    flag_bonds,
+    verify_compatibility,
+)
+
+
+def test_identify_anchors():
+
+    mol = Chem.MolFromSmiles("C1CCC1N")
+    core = [0, 1, 2, 3]
+    anchors = identify_anchors(mol, core, dummy_atom=4)
+    assert set(anchors) == set([3])
+
+    mol = Chem.MolFromSmiles("C1CC2NC2C1")
+    core = [0, 1, 2, 4, 5]
+    anchors = identify_anchors(mol, core, dummy_atom=3)
+    assert set(anchors) == set([2, 4])
+
+    mol = Chem.MolFromSmiles("C1OCC11CCCCC1")
+    core = [3, 4, 5, 6, 7, 8]
+    anchors = identify_anchors(mol, core, dummy_atom=0)
+    assert set(anchors) == set([3])
+    anchors = identify_anchors(mol, core, dummy_atom=1)
+    assert set(anchors) == set([3])
+    anchors = identify_anchors(mol, core, dummy_atom=2)
+    assert set(anchors) == set([3])
+
+    mol = Chem.MolFromSmiles("C1CC1.C1CCCCC1")
+    core = [3, 4, 5, 6, 7, 8]
+    anchors = identify_anchors(mol, core, dummy_atom=0)
+    assert set(anchors) == set()
+    anchors = identify_anchors(mol, core, dummy_atom=1)
+    assert set(anchors) == set()
+    anchors = identify_anchors(mol, core, dummy_atom=2)
+    assert set(anchors) == set()
+
+    mol = Chem.MolFromSmiles("C1CC2NC2C1")
+    core = [0, 1, 2, 5]
+    anchors = identify_anchors(mol, core, dummy_atom=3)
+    assert set(anchors) == set([2, 5])
+    anchors = identify_anchors(mol, core, dummy_atom=4)
+    assert set(anchors) == set([2, 5])
```

Comment:
Should this test have a case where the anchor is made up of 3 atoms?

Reply:
This was confusing to the reader, I've renamed `identify_anchors` to `identify_root_anchors`, defined as the "starting" anchor in an anchor group. A case of a dummy atom connected 3 rooted anchors is extremely rare, but I'll add it for shits an giggles.

Reply:
added this:

```
    mol = Chem.MolFromSmiles("C1C2CC3CC1N23")
    core = [0, 1, 2, 3, 4, 5]
    anchors = identify_root_anchors(mol, core, dummy_atom=6)
    assert set(anchors) == set([1, 3, 5])
```